natss/pkg/dispatcher/dispatcher.go

```diff
@@ -300,6 +374,57 @@ func (s *SubscriptionsSupervisor) subscribe(channel eventingchannels.ChannelRefe
 	return &natssSub, nil
 }
 
+func toEvent(msg *stan.Msg) (*cloudevents.Event, error) {
+	var payload map[string]interface{}
+	if err := json.Unmarshal(msg.Data, &payload); err != nil {
+		return nil, fmt.Errorf("failed to unmarshal message: %+v", err)
+	}
+	event := cloudevents.NewEvent()
+	event.SetSubject(msg.Subject)
+	for k, v := range payload {
```

Comment:
like all of this goes away if you are using the marshalers from cloudevents

Reply:
This piece of code is outdated.